core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -305,7 +325,7 @@ class ReplicaManager(val config: KafkaConfig,
                     responseCallback: Map[TopicPartition, PartitionResponse] => Unit) {
 
     if (isValidRequiredAcks(requiredAcks)) {
-      val sTime = time.milliseconds
+      val sTime = if (logger.isDebugEnabled) time.milliseconds else 0
```

Comment:
I'm a bit confused about the discussion here, so I'll just explain how it works. :) Whatever one passes as a parameter to `debug` is only evaluated if debug logging is enabled. At runtime, a `Function0` is passed and the body of that function is the block of code passed as the parameter to `debug`. Does that make sense?